actionpack/lib/action_controller/metal/rescue.rb

```diff
@@ -6,17 +6,6 @@ module Rescue
     extend ActiveSupport::Concern
     include ActiveSupport::Rescuable
 
-    def rescue_with_handler(exception)
-      if exception.cause
-        handler_index = index_of_handler_for_rescue(exception) || Float::INFINITY
-        cause_handler_index = index_of_handler_for_rescue(exception.cause)
-        if cause_handler_index && cause_handler_index <= handler_index
-          exception = exception.cause
-        end
-      end
-      super(exception)
-    end
-
```

Comment:
Note this exception/cause priority confusion is gone now too. We match the exception to a handler first and only fall back to the cause if none matched.